tsinfer/inference.py

```diff
@@ -1553,20 +1593,41 @@ def get_samples_tree_sequence(self, map_additional_sites=True):
         the non-inference sites also need to be placed into the resulting tree sequence.
         """
         tsb = self.tree_sequence_builder
-
         tables = self.ancestors_ts_tables.copy()
-        num_ancestral_individuals = len(tables.individuals)
 
-        # Currently there's no information about populations etc stored in the
-        # ancestors ts.
+        schema = self.sample_data.metadata_schema
+        tables.metadata_schema = tskit.MetadataSchema(schema)
+        tables.metadata = self.sample_data.metadata
+
+        schema = self.sample_data.populations_metadata_schema
+        if schema is not None:
```

Comment:
We need the "if" here because we want to write the data as raw bytes rather than pass it in as a dictionary. It's ugly, but it was the best way I could think of to provide a path forward without breaking too much code.